### PR TITLE
Added timer to explore v2 and share it with sqllab

### PR DESCRIPTION
**Pull request by vera-liu**
Issue: [https://github.com/airbnb/superset/issues/1563](url)

Done:
 - Share Timer component between sqllab and explore v2

After:
giphy 23

Todo:
 - solve conflicts after [https://github.com/airbnb/superset/pull/1788](url) is merged

needs-review @ascott @mistercrunch @bkyryliuk

### CODE REVIEW

**Comment by mistercrunch** [comment]
The digits look a little big on the gif. Should we style them to be smaller?

File: superset/assets/javascripts/components/Timer.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  endTime: React.PropTypes.number,
+  isRunning: React.PropTypes.bool.isRequired,
+  state: React.PropTypes.string,
+  style: React.PropTypes.object,
+};
+
+Timer.defaultProps = {
+  startTime: null,
+  endTime: null,
+  state: 'success',
```
Comment (ascott) [inline superset/assets/javascripts/components/Timer.jsx]: i think `status` might be more clear in this case, so as not to confuse with the state of the component.

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx [hunk 2]
```diff
@@ -205,6 +214,13 @@ class ChartContainer extends React.Component {
               }
 
               <div className="pull-right">
+                <Timer
```
Comment (ascott) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: could we add some space between the timer and the action buttons to the right? in the screenshot it looks like the are right next to each other. would be nice to add 10px between and make the timer and the buttons aligned in the middle.

**Comment by ascott** [comment]
thanks for taking this @vera-liu! just a couple comments, but LGTM.

**Pull request merged**